zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```
@@ -3412,6 +3413,115 @@ paths:
         "500":
           $ref: "#/components/responses/InternalServerError"
 
+  /batch-operations/{batchOperationKey}:
+    get:
+      tags:
+        - Batch Operation
+      operationId: getBatchOperation
+      summary: Get Batch Operation for batch operation key
+      description: Get Batch Operation
+      parameters:
+        - name: batchOperationKey
+          in: path
+          required: true
+          description: |
+            The key of the batch operation
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Batch Operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/BatchOperationResponse"
+        "400":
+          description: The provided data is not valid.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "404":
+          description: The signal is not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+
+  /batch-operations/{batchOperationKey}/items:
+    get:
+      tags:
+        - Batch Operation Items
+      operationId: getBatchOperationItems
+      summary: Get Items for Batch Operation
+      description: Get Batch Operation Items
+      parameters:
+        - name: batchOperationKey
+          in: path
+          required: true
+          description: |
+            The key of the batch operation
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Batch Operation Items
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/BatchOperationItemSearchQueryResult"
+        "400":
+          description: The provided data is not valid.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "404":
+          description: The signal is not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+
+  /batch-operations/search:
+    post:
+      tags:
+        - Batch Operation
+      operationId: searchBatchOperations
+      summary: Search Batch Operations
+      description: Searches for Batch Operations
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProcessInstanceSearchQuery"
+      responses:
+        "200":
+          description: List of Batch Operations.
```

```suggestion
          description: The batch operation search result.
```

🔧 Aligns with other search endpoints. 
💭 I generally prefer for us to describe the 200 status rather than the response shape, but these search endpoints are the one place I haven't figured out what that should look like.